jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -121,14 +121,14 @@ def process(self, input, *args, **kwargs):
         for key in ['naxis1', 'naxis2']:
             del fit_sip_hdr[key]
 
-        # update meta.wcs_info with fit keywords
-        result.meta.wcsinfo.instance.update(fit_sip_hdr)
-
         # delete naxis, cdelt, pc from wcsinfo
         rm_keys = ['naxis', 'cdelt1', 'cdelt2',
                    'pc1_1', 'pc1_2', 'pc2_1', 'pc2_2']
         for key in rm_keys:
             if key in result.meta.wcsinfo.instance:
                 del result.meta.wcsinfo.instance[key]
 
+        # update meta.wcs_info with fit keywords
+        result.meta.wcsinfo.instance.update(fit_sip_hdr)
```

Comment:
In this case we still end up with `naxis` in `meta.wcsinfo`. I think if you add it to the list of attributes to be remove in line 121 this should fix it.